### PR TITLE
chore: Split related changes for export mi

### PR DESCRIPTION
**Pull request by nidhi-nair**
Splits app json type and a few other refactors as comments from [this PR](https://github.com/appsmithorg/appsmith-ee/pull/3054)

## Summary by CodeRabbit

- **Refactor**
  - Renamed methods for consistency and clarity in action collection services.
  - Streamlined the inheritance and structure of application JSON classes.
  - Improved access modifiers for export services to enhance extensibility.
  - Updated variable naming for clarity in import services.
  - Enhanced method signatures in repositories to better handle permissions.

- **Documentation**
  - Updated method names and parameters in documentation to reflect refactoring changes.

- **Chores**
  - Cleaned up deprecated fields and updated method calls to align with the new method names.

### CODE REVIEW

**Review by subrata71 — APPROVED** [review]

File: app/server/appsmith-server/src/main/java/com/appsmith/server/newactions/imports/NewActionImportableServiceCEImpl.java [hunk 1]
```diff
@@ -262,7 +262,7 @@ private Mono<ImportActionResultDTO> importActions(
 
                                 for (NewAction newAction : importedNewActionList) {
                                     if (newAction.getUnpublishedAction() == null
-                                            || !org.springframework.util.StringUtils.hasLength(newAction
+                                            || !StringUtils.hasLength(newAction
```
[the remark this answers is not in the document]
Comment (nidhi-nair) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/newactions/imports/NewActionImportableServiceCEImpl.java:265]: Incorrect suggestion

File: app/server/appsmith-server/src/main/java/com/appsmith/server/jslibs/exports/CustomJSLibExportableServiceCEImpl.java [hunk 2]
```diff
@@ -38,12 +38,7 @@ public Mono<Void> getExportableEntities(
          * Since we are exporting for git, we only consider unpublished JS libraries
          * Ref: https://theappsmith.slack.com/archives/CGBPVEJ5C/p1672225134025919
```
[the remark this answers is not in the document]
Comment (nidhi-nair) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/jslibs/exports/CustomJSLibExportableServiceCEImpl.java:39]: Irrelevant for this PR

File: app/server/appsmith-server/src/main/java/com/appsmith/server/dtos/ce/ApplicationJsonCE.java [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+     * changes in database and the last local git commit.
+     * This field can be used while saving resources to local file system and only update the resource files which
+     * are updated in the database.
+     */
+    @JsonView(Views.Internal.class)
+    Map<String, Set<String>> updatedResources;
+
+    // TODO remove the plain text fields during the export once we have a way to address sample apps DB authentication
+    @JsonView(Views.Public.class)
+    Map<String, DecryptedSensitiveFields> decryptedFields;
```
[the remark this answers is not in the document]
Comment (nidhi-nair) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/dtos/ce/ApplicationJsonCE.java:92]: Irrelevant for this PR

File: app/server/appsmith-server/src/main/java/com/appsmith/server/newactions/imports/NewActionImportableServiceCEImpl.java [hunk 4]
```diff
@@ -291,8 +291,7 @@
 
                                     if (publishedAction != null && publishedAction.getValidName() != null) {
                                         publishedAction.setId(newAction.getId());
-                                        if (!org.springframework.util.StringUtils.hasLength(
-                                                publishedAction.getPageId())) {
+                                        if (!StringUtils.hasLength(publishedAction.getPageId())) {
```
[the remark this answers is not in the document]
Comment (nidhi-nair) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/newactions/imports/NewActionImportableServiceCEImpl.java:294]: Incorrect suggestion

**Pull request merged**